src/System.Private.CoreLib/shared/System/Globalization/StringInfo.cs

```diff
@@ -2,103 +2,69 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-////////////////////////////////////////////////////////////////////////////
-//
-//
-//  Purpose:  This class defines behaviors specific to a writing system.
-//            A writing system is the collection of scripts and
-//            orthographic rules required to represent a language as text.
-//
-//
-////////////////////////////////////////////////////////////////////////////
-
-using System;
 using System.Diagnostics;
 
 namespace System.Globalization
 {
+    /// This class defines behaviors specific to a writing system.
+    /// A writing system is the collection of scripts and orthographic rules
+    /// required to represent a language as text.
     public class StringInfo
     {
         private string _str;
 
         private int[] _indexes;
 
-        // Legacy constructor
-        public StringInfo() : this("") { }
-
-        // Primary, useful constructor
-        public StringInfo(string value)
+        public StringInfo() : this(string.Empty)
         {
-            this.String = value;
         }
 
-        public override bool Equals(object value)
+        public StringInfo(string value)
         {
-            if (value is StringInfo that)
-            {
-                return (_str.Equals(that._str));
-            }
-            return (false);
+            String = value;
```

Comment:
>String [](start = 12, length = 6)

I think keeping 'this' here is a little bit better to distinguish it from String type

Reply:
Fair, fixed